src/main/scala/com/typesafe/sbt/packager/archetypes/jlink/JlinkPlugin.scala

```diff
@@ -100,16 +100,30 @@ object JlinkPlugin extends AutoPlugin {
         sys.error("Missing package dependencies")
       }
 
-      // Collect all the found modules
-      deps.collect {
+      val detectedModuleDeps = deps.collect {
         case PackageDependency(_, _, PackageDependency.Module(module)) =>
           module
-      }.distinct
+      }.toSet
+
+      // Some JakartaEE artifacts use `java.*` module names, even though
+      // they are not a part of the platform anymore.
+      // https://github.com/eclipse-ee4j/ee4j/issues/34
+      // This requires special handling on our part when deciding if the module
+      // is a part of the platform or not.
+      // At least the new modules shouldn't be doing this...
+      val knownJakartaJavaModules = Set("java.xml.bind", "java.xml.soap", "java.ws.rs")
```

Comment:
I'm curious if this is something we should make configurable if there are more of these special cases?

Reply:
Users can filter `jlinkModules` to achieve that, so we don't need another setting.

Reply:
Thanks for the clarification :+1: